llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -545,15 +545,12 @@ void AIEPostRASchedStrategy::enterMBB(MachineBasicBlock *MBB) {
 
 void AIEPostRASchedStrategy::commitBlockSchedule(MachineBasicBlock *BB) {
   auto &BS = InterBlock.getBlockState(BB);
-  MachineBasicBlock::iterator It = BB->begin();
-  const TargetInstrInfo *TII = getTII(BB);
 
-  // Emit the top safety margin
-  int Margin = InterBlock.getNumEntryNops(BS);
-  while (Margin--) {
-    TII->insertNoop(*BB, It);
-  }
+  // Safety margin, swp epilogue
+  InterBlock.emitInterBlockTop(BS);
```

Comment:
We emit epilogue-specific code, shouldn't that be in `emitInterBlockBottom`?

Reply:
The epilogue is pushed into the top of the exit block.

Reply:
I get that, but reading the function name really sounds like "we emit the top of the interblock schedule". I think emitInterblockPrologue and emitInterblockEpilogue is clearer. But nvm, I won't block for that.